File: sample/client/proto/Forum.ForumData.cs
// sample/proto/Forum.proto
using ILib.ProtoPack;
using System.Collections.Generic;

namespace Forum
{
	[System.Serializable]
	public partial class ForumData : IMessage
	{
		public Forum.PostData[] Data;

		#region Serialization

		/// <summary>
		/// Serialize Message
		/// </summary>
		public void Write(IWriter w)
		{
			// Write Map Length
			w.WriteMapHeader(1);
			
			// Write data
			w.WriteTag(1);
			if (Data == null)
			{
				w.WriteNil();
			}
			else
			{
				var arrayLen = Data.Length;
				w.WriteArrayHeader(arrayLen);
				for(var arrayIndex = 0; arrayIndex < arrayLen; arrayIndex++)
				{
					w.Write(Data[arrayIndex]);
				}
			}
		}

		/// <summary>
		/// Deserialize Message
		/// </summary>
		public void Read(IReader r)
		{
			// Read Map Length
			var len = r.ReadMapHeader();

			for (var i = 0; i < len; i++)
			{
				var tag = r.ReadTag();
				switch(tag) {
				case 1: // Read data
					if(r.NextFormatIsNull())
					{
						r.ReadNil();
						this.Data = null;
						continue;
					}
					var _DataLen = r.ReadArrayHeader();
					Data = new Forum.PostData[_DataLen];
					for(int arrayIndex = 0; arrayIndex < _DataLen; arrayIndex++)
					{
						Data[arrayIndex] = r.ReadMessage<Forum.PostData>();
					}
					break;
				default:
					r.Skip();
					break;
				}
			}
		}
		#endregion

	}
}

File: sample/goclient/proto/forum.forumdata.go
// sample/proto/Forum.proto

package proto

import (
	protopack "github.com/yazawa-ichio/protoc-gen-msgpack/lib/golang"
)

type Forum_ForumData struct {
	Data []*Forum_PostData
}

// Pack Serialize Message
func (m *Forum_ForumData) Pack() ([]byte, error) {
	return protopack.Pack(m)
}

// Write Serialize Message
func (m *Forum_ForumData) Write(w protopack.Writer) error {
	// Write Map Length
	err := w.WriteMapHeader(1)
	if err != nil {
		return err
	}
	
	// Write data
	err = w.WriteTag(1)
	if err != nil {
		return err
	}
	if m.Data == nil{
		err = w.WriteNil()
		if err != nil {
			return err
		}
	} else {
		arrayLen := len(m.Data)
		err = w.WriteArrayHeader(arrayLen)
		if err != nil {
			return err
		}
		for arrayIndex, _ := range m.Data {
			err = w.WriteMessage(m.Data[arrayIndex])
			if err != nil {
				return err
			}
		}
	}
	return err
}

// Unpack Serialize Message
func (m *Forum_ForumData) Unpack(buf []byte) (error) {
	return protopack.Unpack(m, buf)
}

// Read Deserialize Message
func (m *Forum_ForumData) Read(r protopack.Reader) error {
	// Read Map Length
	len, err := r.ReadMapHeader()
	if err != nil {
		return err
	}

	for i := uint(0); i < len; i++ {

		// Read Tag
		tag, err := r.ReadTag()
		if err != nil {
			return err
		}

		switch tag {
		case 1: // Read data
			isArrayNil, err := r.NextFormatIsNull() 
			if err != nil {
				return err
			}
			if isArrayNil {
				r.ReadNil()
				m.Data = nil
				continue
			}
			_DataLen, err := r.ReadArrayHeader();
			if err != nil {
				return err
			}
			m.Data = make([]*Forum_PostData, _DataLen);
			for arrayIndex := uint(0); arrayIndex < _DataLen; arrayIndex++ {
				isNil, err := r.NextFormatIsNull()
				if err != nil {
					return err
				}
				if isNil {
					m.Data[arrayIndex] = nil
					err = r.ReadNil()
				} else {
					m.Data[arrayIndex] = &Forum_PostData{}
					err = m.Data[arrayIndex].Read(r)
				}
				if err != nil {
					return err
				}
			}
			break
		default:
			err = r.Skip()
			if err != nil {
				return err
			}
			break
		}
	}
	return err
}
